frontend/dialob-composer-material/src/views/tree/NavigationTreeView.tsx

```diff
@@ -0,0 +1,109 @@
+import React, { useState } from 'react';
+import { Box, useTheme } from '@mui/material';
+import Tree, {
+  mutateTree,
+  moveItemOnTree,
+  TreeData,
+  ItemId,
+  TreeSourcePosition,
+  TreeDestinationPosition,
+  RenderItemParams,
+} from '@atlaskit/tree';
+import { useComposer } from '../../dialob';
+import { buildTreeFromForm } from './TreeBuilder';
+import NavigationTreeItem from './NavigationTreeItem';
+
+
+const isParentNode = (tree: TreeData, destination?: TreeDestinationPosition): boolean => {
+  if (!destination) {
+    return false;
+  }
+  if (destination.parentId.toString().includes('root')) {
+    return true;
+  }
+  const destinationItem = tree.items[destination.parentId];
+  if (destinationItem.data.type === 'group' || destinationItem.data.type === 'surveygroup' || destinationItem.data.type === 'rowgroup') {
+    return true;
+  }
+  return false;
+}
+
+const isEmptyParentNode = (tree: TreeData, destination?: TreeDestinationPosition): boolean => {
+  const isParent = isParentNode(tree, destination);
+  if (!isParent) {
+    return false;
+  }
+  const destinationItem = tree.items[destination!.parentId];
+  return destinationItem.children.length === 0;
+}
+
+const renderItem = ({ item, onExpand, onCollapse, provided }: RenderItemParams) => {
+  return (
+    <NavigationTreeItem item={item} onExpand={onExpand} onCollapse={onCollapse} provided={provided} />
+  );
+}
+
+const NavigationTreeView: React.FC = () => {
+  const theme = useTheme();
+  const { form, moveItem } = useComposer();
+  const treeData: TreeData = buildTreeFromForm(form.data);
```

Comment:
If this `treeData` is used here only for the initial value given to `useState` below, please put it to useState as a callback function: `useState<TreeData>(() => buildTreeFromForm(....))`
Doing it like this will ensure that the potentially expensive function is called only once, not on every render.

At the moment, the function is always called on every render here even though the value is ignored by useState if it is already set.

Reply:
Now that I look at it better, the `useEffect` will take care of building the tree, so just a skeleton of the tree structure could be passed as the initial value so the `buildTreeFromForm` function is called one less time, [like so](https://github.com/dialob/dialob-parent/pull/71/commits/dfda6cbc39c5a53dd7b2ed3af3eac993e8974fd5#)